cibuildwheel/__main__.py

```diff
@@ -107,7 +107,7 @@ def main() -> None:
         help="Enable pre-release Python versions if available.",
     )
 
-    args = parser.parse_args(namespace=CommandLineArguments())
```

Comment:
I guess this'll give us a runtime failure if the args don't match the object, which would help us catch mismatches.